src/main/java/com/google/impactdashboard/database_manager/bigquery/QueryConfigurationBuilderImpl.java

```diff
@@ -0,0 +1,160 @@
+package com.google.impactdashboard.database_manager.bigquery;
+
+import com.google.cloud.bigquery.QueryJobConfiguration;
+
+/** 
+ * A class for building query configuration builder objects and storing them, 
+ * so that they only need ot be built once. The query job configuration builder 
+ * objects will be configured to query the database with real project information. 
+ */
+public class QueryConfigurationBuilderImpl implements QueryConfigurationBuilder {
+  private static final QueryConfigurationBuilderImpl INSTANCE = new QueryConfigurationBuilderImpl();
+  
+  private QueryJobConfiguration.Builder getProjectIdsConfiguration = null;
+  private QueryJobConfiguration.Builder getProjectIdentificationInformationConfiguration = null;
+  private QueryJobConfiguration.Builder getAverageBindingsConfiguration = null;
+  private QueryJobConfiguration.Builder getDatesToBindingsConfiguration = null;
+  private QueryJobConfiguration.Builder getDatesToIAMRecommendationsConfiguration = null;
+  private String insertValuesIAMTableConfiguration = null;
+  private String insertValuesRecommendationsTableConfiguration = null;
+  private QueryJobConfiguration.Builder deleteOldDataIAMTableConfiguration = null;
+  private QueryJobConfiguration.Builder deleteOldDataRecommendationsTableConfiguration = null;
+
+  private QueryConfigurationBuilderImpl() { }
+
+  /** Returns the only instance of the class. */
+  public static QueryConfigurationBuilder getInstance() {
+    return INSTANCE;
+  }
+
+  /** 
+   * Retrieves query job configuration that retrieves all project ids from the 
+   * database. 
+   */
+  public QueryJobConfiguration.Builder getProjectIdsConfiguration() {
+    if (getProjectIdsConfiguration == null) {
+      getProjectIdsConfiguration = QueryJobConfiguration
+        .newBuilder(Queries.GET_PROJECT_IDS).setUseLegacySql(false);
+    }
+
```

Comment:
Could you clarify what you mean by this? I don't think I understand what you're asking. Do you want me to declare all the fields `static final`? I'm not sure if I can since the builder objects need to be modified (to add the parameters) . Or do you want me to just use 
```
  private QueryJobConfiguration.Builder getProjectIdsConfiguration = QueryJobConfiguration
        .newBuilder(Queries.GET_PROJECT_IDS).setUseLegacySql(false);
```
instead of 
```
  private QueryJobConfiguration.Builder getProjectIdsConfiguration = null;
```